apps/easypid/src/features/receive/FunkeOpenIdCredentialNotificationScreen.tsx

```diff
@@ -0,0 +1,146 @@
+
+  const [credentialRecord, setCredentialRecord] = useState<W3cCredentialRecord | SdJwtVcRecord>()
+  const [errorReason, setErrorReason] = useState<string>()
+  const [isAccepted, setIsAccepted] = useState(false)
+  const [isStoring, setIsStoring] = useState(false)
+
+  const credential = credentialRecord ? getCredentialForDisplay(credentialRecord) : undefined
+
+  useEffect(() => {
+    const requestCredential = async (params: Query) => {
+      try {
+        const { resolvedCredentialOffer } = await resolveOpenId4VciOffer({ agent, offer: params })
+        const tokenResponse = await acquireAccessToken({ agent, resolvedCredentialOffer })
+        const [credentialRecord] = await receiveCredentialFromOpenId4VciOffer({
+          agent,
+          resolvedCredentialOffer,
+          accessToken: tokenResponse,
+        })
+
+        if (credentialRecord.type === 'MdocRecord') {
+          throw new Error('mdoc not supported')
+        }
+        setCredentialRecord(credentialRecord)
+      } catch (e: unknown) {
+        agent.config.logger.error(`Couldn't receive credential from OpenID4VCI offer`, {
+          error: e,
+        })
+        setErrorReason('Credential information could not be extracted')
+      }
+    }
+    void requestCredential(params)
+  }, [params, agent])
+
+  const onCredentialAccept = async () => {
+    if (!credentialRecord) return
+
+    setIsStoring(true)
+    await storeCredential(agent, credentialRecord)
+      .then(async () => {
+        const { metadata, display } = getCredentialForDisplay(credentialRecord)
+
+        await activityStorage.addActivity(agent, {
+          id: credentialRecord.id,
+          type: 'received',
+          date: new Date().toISOString(),
+          entityHost: getHostNameFromUrl(metadata.issuer) as string,
+          entityName: display.issuer.name,
+        })
+
+        setIsAccepted(true)
+      })
+      .catch((e) => {
+        agent.config.logger.error("Couldn't store credential", {
+          error: e as unknown,
+        })
+        setErrorReason('Couldn’t store credential')
+      })
+  }
+
+  const onCredentialDecline = () => {
+    toast.show('Credential has been declined.')
+    pushToWallet('back')
+  }
+
+  const onCancel = () => pushToWallet('back')
+  const onComplete = () => pushToWallet('replace')
+
+  return (
+    <SlideWizard
+      steps={[
+        {
+          step: 'loading-request',
+          progress: 16.5,
+          screen: <LoadingRequestSlide key="loading-request" isLoading={!credentialRecord} isError={!!errorReason} />,
+        },
+        {
+          step: 'verify-issuer',
+          progress: 33,
+          backIsCancel: true,
+          screen: (
+            <VerifyPartySlide
+              key="verify-issuer"
+              name={credential?.display.issuer.name}
+              logo={credential?.display.issuer.logo}
+              // @ts-expect-error issuer is not typed
+              domain={credentialRecord?.metadata.issuer as string}
```

Comment:
We have the openid4vc metadata, we should use util methods to get / set this. As this will cause bugs. We need an unified way to store metadata for a cred

Reply:
I've removed the use of metadata values